doc/src/tasks.adoc

```diff
@@ -726,3 +726,98 @@ managers. Currently the supported ones are:
 
 * Conan (`conan`): currently supports the
   link:https://docs.conan.io/en/latest/reference/generators/b2.html[`b2 generator`].
+
+[[bbv2.tasks.projectsearch]]
+== Searching For Projects
+
+B2 supports automatic searching for referenced global projects. For example,
+if you have references to `/boost/predef` with some minimal configuration B2
+can find the B2 project for it and automatically resolve the reference. The
+searching supports two modes: finding regular B2 project directories, and
+package/config style loading of single jam files.
+
+[[bbv2.tasks.projectsearch.path]]
+=== Search Path
+
+To control which and where projects are found one can use different methods:
+
+* `B2_PROJECT_PATH` environment variable.
+* `--project-search` command line argument.
+* `rule project-search` project rule.
+
+The search path in both `B2_PROJECT_PATH` and `--project-search` specifies a
+key-value list of _project-id_ and _path_. The parts of that key-value list, as
```

Comment:
The project (Boost) that has Jamroot will eventually include everything in its subdirectories

> he use case I'm working on ATM is the case of a modular Boost. Where definitely not all projects would be loaded even though they could be under the same specified path.

Can't Boost superproject (Jamroot) have that loader which will load a needed subproject when you request a library? Or your vision is that every Boost library will get its own Jamroot and libraries will no longer will be rooted to the superproject? But then I don't understand who and how will set and update these mappings for every library.

Reply:
I see what I have missed: `libs` part in `--project-search=/boost:/*/boostroot/libs`, which looks so strange and alien while still not working for sublibs and stuff in `tools` folder.

>  [grafikrobot/boost-b2-modular@`b2-modular`/patch/libs/serialization/build.jam#L8](https://github.com/grafikrobot/boost-b2-modular/blob/b2-modular/patch/libs/serialization/build.jam?rgh-link-date=2023-06-07T01%3A48%3A21Z#L8).

I'm sorry to say that but I have to -- it looks ugly, writing the same stuff twice... Is it impossible to make `<library>/boost/array` work? Even better if it could be just simple `/boost/array`. It probably could be achieved by considering project-id reference as implying adding every project top-level target as `<source>`.